### DISCUSSION TITLE
Mapping data of architectural plan to mesa

### DISCUSSION THREAD

**Discussion by AlABarazi** [mesa/mesa-geo#151] [Q&A] [answered]
Hi,
I have architecture plan containing drywalls agents (with unique ID, amount of work needed for each trade (panel, electrical, mechanical) ) , external walls, staircases, spaces(apartments).   The model also should have construction workers agents. What would be a better way to map such data to mesa? Do you recommend using grids or geospace? And if you recommend to use geospace, which tools can be used to build it (connect the data to the the architecture plan).

**Comment by wang-boyu** [comment, marked as answer]
Hi!

I guess the question is about whether to use discrete space or continuous space? In Mesa, there are grid space and continuous space; in geospace from Mesa-Geo, there are raster layer of cells (grid-like) and vector layer of geoagents (continuous space-like).

To be honest, I've never worked with any architecture plan before. From the description, it seems that drywall agents have their own unique ids and shapes. So if you use Mesa-Geo, save these agents as GeoDataFrames and create them as GeoAgents might be a good starting point. Raster layer of cells is also possible.

But (I'm not sure about this) perhaps the architecture plan are not geo-tagged? That is, do drywall agents and construction worker agents have their latitude/longitude values? If no, then you might want to use Mesa to build your model. Depending on the type of their coordinates (discrete or continuous), you may choose to use grid space or continuous space from Mesa accordingly.
